modules/programs/readline.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.readline;
+in
+{
+  options.programs.readline = {
+    enable = mkEnableOption "readline";
+
+    bindings = mkOption {
+      default = {};
+      type = types.attrsOf types.str;
+      example = { "\C-h" = "backward-kill-word"; };
+      description = "Readline bindings";
+    };
+
+    includeSystem = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to include the system-wide config";
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Config lines appended to the end unchanged. Useful for configuration not yet supported by home-manager.";
+    };
+  };
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.readline ];
```

Comment:
Is it really necessary to install the readline package? It only provides a few libraries which makes it a pretty unconventional package to explicitly install into your profile…

Reply:
Honestly, I am a total Nix/home-manager beginner, so any advice is appreciated.
Without readline, `~/.inputrc` is pointless, so I put it there since other home-manager modules like vim, do so as well.

Reply:
Yeah, but in these other cases, like vim, the package provides an executable binary that we expect the users wanting to run. The readline package, on the other hand, only provides libraries. For the Nix package manager it is not necessary to install pure library packages into the user profile.

This is because any program that uses readline will have been linked with a specific version of readline and will refer directly to that version inside the Nix store. For example,
```console
$ ldd $(which bash) | grep readline
	libreadline.so.7 => /nix/store/n0ssjcd7czbrbz64xm6rn9qcxw65dp8f-readline-7.0p5/lib/libreadline.so.7 (0x00007f0849c3c000)
	libhistory.so.7 => /nix/store/n0ssjcd7czbrbz64xm6rn9qcxw65dp8f-readline-7.0p5/lib/libhistory.so.7 (0x00007f0849c2f000)
```

Reply:
Thanks for the explanation, I understand now and agree. I will remove the package installation.